scripts/meta/west/main.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2018 Open Source Foundries Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Zephyr RTOS meta-tool (west) main module
+'''
+
+
+import argparse
```

Comment:
I did a little click evaluation a while ago when we were first starting this up. I don't remember the specifics, but I found its convenience came at the cost of it being a bit too opinionated in some cases.

I am happy to discuss this more a bit further down the line though if click has compelling benefits, though.

(FWIW, I've used somewhat-tricky argparse features in this body of code, and the `runner` package already uses `argparse`, so a conversion effort could be non-trivial.)

Reply:
I agree that while using click in other projects I did feel constrained a little bit by the decisions taken by the module author. That said, it does save quite a bit of boilerplate code. I'll see if I can check if the constraints I talk about have been addressed since the last time I used it.